discovergy/discovergy.py

```diff
@@ -42,14 +42,20 @@ def _fetch_consumer_tokens(self):
                                      data={'client': self._client_name},
                                      headers={},
                                      timeout=TIMEOUT)
+        except Exception as e:
+            logger.error('Failed consumer token request: %s', e)
```

Comment:
Please return or retry after a limited number of attempts after a failed consumer token request.